crates/api/src/trap.rs

```diff
@@ -12,8 +12,27 @@ pub struct Trap {
     inner: Arc<TrapInner>,
 }
 
+/// State describing the occasion which evoked a trap.
+#[derive(Debug)]
+pub enum TrapReason {
+    /// An error message describing a trap.
+    Message(String),
```

Comment:
Exposing an `enum` often makes pretty hard guarantees about representations which are tough to change. Could we  perhaps retain `fn message()` with a canned string for exit statuses, and then add an extra accessor returning `Option<i32>` for reading the exit code?

Reply:
Done